clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3444,17 +3468,16 @@ bool Util::isSyclSpecConstantType(const QualType &Ty) {
 }
 
 bool Util::isPropertyListType(const QualType &Ty) {
-  return isSyclType(Ty, "property_list", true /*Tmpl*/);
+  return isSyclType(Ty, "property_list");
 }
 
 bool Util::isSyclBufferLocationType(const QualType &Ty) {
   const StringRef &Name = "buffer_location";
-  std::array<DeclContextDesc, 4> Scopes = {
+  std::array<DeclContextDesc, 6> Scopes = {
       Util::DeclContextDesc{clang::Decl::Kind::Namespace, "cl"},
       Util::DeclContextDesc{clang::Decl::Kind::Namespace, "sycl"},
-      // TODO: this doesn't belong to property namespace, instead it shall be
-      // in its own namespace. Change it, when the actual implementation in SYCL
-      // headers is ready
+      Util::DeclContextDesc{clang::Decl::Kind::Namespace, "ext"},
+      Util::DeclContextDesc{clang::Decl::Kind::Namespace, "INTEL"},
```

Comment:
I find this nesting a little bit confusing - addressed the question in https://github.com/intel/llvm/pull/2439/files#r485446113